### PR TITLE
Update infrastructure page

### PR DESCRIPTION
**Pull request by ehuss**
This is #144 rebased, and then with a bunch of my own updates.

### CODE REVIEW

File: infrastructure.md [hunk 1]
```diff
@@ -38,48 +40,87 @@ goes wrong with the bot.
...
+
+[perf](https://github.com/rust-lang-nursery/rustc-perf) offers information
+about the performance of `rustc` over time.
+
+It is split into a data [collector] and a [web frontend]. The raw performance
+data is available [here](https://github.com/rust-lang-nursery/rustc-timing)
+and can be browsed on the [perf website](https://perf.rust-lang.org).
+
+[collector]: https://github.com/rust-lang-nursery/rustc-perf/tree/master/collector
+[web frontend]: https://github.com/rust-lang-nursery/rustc-perf/tree/master/site
```
Comment (Centril) [inline infrastructure.md]: Could we add a quick blurb about the most common use case and commands to achieve that?

```
@bors try

<wait a bit; copy the $hash from the build>

@rust-timer build $hash
```

File: infrastructure.md [hunk 2]
```diff
@@ -38,48 +40,87 @@ goes wrong with the bot.
...
+versions of the Rust compiler.
+
+## docs.rs
+
+[docs.rs](https://docs.rs/) builds and serves the rustdoc documentation for
+all crates on [crates.io]. Issues may be filed on the [docs.rs
+repository](https://github.com/rust-lang/docs.rs). See the #docs-rs channel on
+Discord for discussion or urgent issues.
+
+## Rust Log Analyzer
```
Comment (Centril) [inline infrastructure.md]: Put this below highfive?

File: infrastructure.md [hunk 3]
```diff
@@ -38,48 +40,87 @@ goes wrong with the bot.
...
+The [Rust Log Analyzer](https://github.com/rust-lang/rust-log-analyzer)
+analyzes CI build logs to extract error messages and posts them to the pull
+request. It is run by [TimNN](https://github.com/TimNN).
+
+## Toolstate
+
+The state of tools included with Rust are tracked on the [toolstate
+page](https://rust-lang-nursery.github.io/rust-toolstate/). When each PR is
+merged via CI, the status of each tool is recorded in a JSON file and stored
+in the [toolstate repo](https://github.com/rust-lang-nursery/rust-toolstate).
```
Comment (Centril) [inline infrastructure.md]: Say something about the toolstate week and when it happens? cc @pietroalbini

Comment (ehuss) [reply infrastructure.md]: Toolstate is being documented in #243.  This can link to that if/when that gets merged.

Comment (Centril) [reply infrastructure.md]: Alright; sounds good.

File: infrastructure.md (hunk 1 above)
Comment (ehuss) [reply infrastructure.md]: I've never used the rust-timer bot, but I added a small blurb.  Ideally I would think documentation should be on the perf website (I searched around and couldn't find any anywhere, like who has permissions to use it, etc.).

Comment (Centril) [reply infrastructure.md]: cc @Mark-Simulacrum

**Comment by Centril** [comment]
I'm happy with this for the time being; we can make improvements if need be later.

File: infrastructure.md (hunk 3 above)
Comment (RalfJung) [reply infrastructure.md]: I added a link from here to the new page in #243.

**Pull request merged**